internal/controller/compaction/utils.go

```diff
@@ -0,0 +1,51 @@
+package compaction
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	druidmetrics "github.com/gardener/etcd-druid/internal/metrics"
+
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type httpClientInterface interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func fullSnapshot(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, httpClient httpClientInterface, httpScheme string) error {
```

Comment:
why is this function in utils.go? Is it used anywhere other than compaction?
instead create a new file and name it `snapshot.go` and move all functions related to taking a snapshot there.